frontend/src/book/store.js

```diff
@@ -131,16 +131,34 @@ export const setPageIndex = (pageIndex) => {
 };
 
 export const movePage = (delta) => {
+  if (delta > 0) {
+    animatePageTransition();
+  }
   setPageIndex(currentPage().index + delta);
 };
 
+const animatePageTransition = () => {
+  document.body.style.animation = 'bodyTransition 0.5s linear';
+  setTimeout(() => document.body.style.animation = '', 500);
+};
+
 export const moveStep = (delta) => {
   const stepIndex = currentStep().index + delta;
   const step = currentPage().steps[stepIndex];
   if (!step) {
     return;
   }
+  
   setUserStateAndDatabase(["pagesProgress", localState.user.pageSlug, "step_name"], step.name);
+  if (delta > 0) {
+    animateNextStep(stepIndex);
+  }
+};
+
+const animateNextStep = (stepIndex) => {
+  const div = document.getElementById(`step-text-${stepIndex}`);
```

Comment:
I get an error that the div is null when clicking "Skip step" (see https://github.com/alexmojaki/futurecoder#controls)

Reply:
Should be fixed now. When steps are skipped, the dom doesn't update at the same time, so I added a quick timeout of 1ms and it should now catch it.